src/QuadForm/Lattices.jl

```diff
@@ -1569,3 +1588,37 @@ and which is maximal in `V` satisfying this property.
 """
 maximal_integral_lattice(::AbsSpace)
 
+################################################################################
+#
+#  Saturation
+#
+################################################################################
+
+@doc Markdown.doc"""
+    saturate(L::AbsLat) -> AbsLat
+
+Given a lattice `L` over a field `E`, return the closure $L\otimes E \cap O_E^n$
+of `L` in its rational span (see [`rational_span`](@ref)). Here $O_E$ is the base
+ring of `L` (see [`base_ring`](@ref)) and `n` is the degree of `L`.
+"""
+saturate(L::AbsLat) = _saturate_via_restriction_of_scalars(L)
+
+function _saturate_via_restriction_of_scalars(L::AbsLat)
+  @assert has_ambient_space(L)
+  bool = L isa ZLat
```

Comment:
style wise you would rather define a method
`_saturate_via_restriction_of_scalars(L::ZLat)` 
instead of this kind of branching inside the method. It is more transparent when it is part of the signature.

Reply:
I see but the problem is that I am afraid of "ambiguity" because formally, on Hecke, `ZLar` is a subtype of `AbsLat`, so I don't want to create ambiguity leading to problems having the method for `ZLat` and `AbsLat`. And if there are no ambiguity problems, then I don't understand how it works for Julia/Hecke to choose the right method to call...

Reply:
This won't lead to an ambiguity since `ZLat <: AbsLat` is more specialized. Julia always looks for the most special method.